src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -0,0 +1,2515 @@
+  if(rhs.type().subtype().id()==ID_symbol)
+    deref_type=symbol_table.lookup(
+      to_symbol_type(rhs.type().subtype()).get_identifier()).type;
+  else
+    deref_type=rhs.type().subtype();
+
+  dereference_exprt deref(rhs, deref_type);
+
+  // Fields of the string object
+  exprt rhs_length=get_length(deref, symbol_table);
+  exprt member_data=get_data(deref, symbol_table);
+  dereference_exprt rhs_data(member_data, member_data.type().subtype());
+
+  // Assignments
+  code_blockt code;
+  code.add(code_assignt(lhs.length(), rhs_length));
+  code.add(code_assignt(lhs.content(), rhs_data));
+  return code;
+}
+
+/*******************************************************************\
+
+Function: java_string_library_preprocesst::
+            code_assign_string_literal_to_string_expr
+
+  Inputs:
+    lhs - an expression representing a java string
+    tmp_string - a temporary string to hold the literal
+    s - the literal to be assigned
+    symbol_table - symbol table
+
+  Output: return the following code:
+          > tmp_string = "s"
+          > lhs = (string_expr) tmp_string
+
+\*******************************************************************/
+
+codet java_string_library_preprocesst::
+  code_assign_string_literal_to_string_expr(
+    const string_exprt &lhs,
+    const exprt &tmp_string,
+    const std::string &s,
+    symbol_tablet &symbol_table)
+{
+  code_blockt code;
+  code.add(code_assignt(tmp_string, string_literal(s)));
+  code.add(code_assign_java_string_to_string_expr(
+    lhs, tmp_string, symbol_table));
+  return code;
+}
+
+/*******************************************************************\
+
+Function: java_string_library_preprocesst::
+    make_string_builder_append_object_code
+
+  Inputs:
+    type - type of the function called
+    loc - location in the program
+    symbol_table - symbol table
+
+ Outputs: code for the StringBuilder.append(Object) function:
+          > string1 = arguments[1].toString()
+          > string_expr1 = string_to_string_expr(string1)
+          > string_expr2 = concat(this, string_expr1)
+          > this = string_expr_to_string(string_expr2)
+          > return this
+
+\*******************************************************************/
+
+codet java_string_library_preprocesst::make_string_builder_append_object_code(
+  const code_typet &type,
+  const source_locationt &loc,
+  symbol_tablet &symbol_table)
+{
+  code_typet::parameterst params=type.parameters();
+  assert(params.size()==1);
+  exprt this_obj=symbol_exprt(params[0].get_identifier(), params[0].type());
+
+  // Code to be returned
+  code_blockt code;
+
+  // String expression that will hold the result
+  string_exprt string_expr1=fresh_string_expr(loc, symbol_table, code);
+
+  exprt::operandst arguments=process_parameters(
+    type.parameters(), loc, symbol_table, code);
+  assert(arguments.size()==2);
+
+  // > string1 = arguments[1].toString()
+  exprt string1=fresh_string(this_obj.type(), loc, symbol_table);
```

Comment:
OK, I now saw @romainbrenguier's comment that this will be split. So I'll look at it again once that is done.